distributed/rpc/ddp_rpc/requirements.txt

```diff
@@ -1 +1 @@
-torch>=1.6.0
+torch==1.9.0
```

Comment:
Can you double-check if we can reenable this test https://github.com/pytorch/examples/blob/main/run_python_examples.sh#L176 and add a small function for it

May have a couple of APIs that have been deprecated so worst case we can just make a documentation only PR

Reply:
I was able to create tests to trigger the method, but because my local machine is a Macbook I don't have access to CUDA. What would you recommend as the best way to test it - my hunch is that it would make most sense to spin up a deep learning image in any given cloud. Thoughts?

Reply:
That's what I do, I usually use one the Ubuntu Deep Learning AMI on AWS, you can choose either a p2 or g4dn instance

Reply:
@veekaybee I'm super sorry but now that I think more about this we'd need to also have a GPU instance in a Github runner as well to run GPU tests on CI so there's no need to add tests but if you could sanity check that that the scripts do run indeed once you upgrade the torch version, maybe all the way to 1.11 then this would still be a valuable contribution

Reply:
I'm getting the following error when testing running the [main method here](https://github.com/pytorch/examples/blob/349c08e95acda442978e83400648afb3e3764036/distributed/rpc/ddp_rpc/main.py#L177) 

```
Running example: distributed
On WorkerInfo(id=1, name=trainer1):
RuntimeError('CUDA error: invalid device ordinal\nCUDA kernel errors might be asynchronously reported at some other API call,so the stacktrace below might be incorrect.\nFor debugging consider passing CUDA_LAUNCH_BLOCKING=1.')
Traceback (most recent call last):
  File "/home/ubuntu/anaconda3/envs/pytorch_p38/lib/python3.8/site-packages/torch/distributed/rpc/internal.py", line 204, in _run_function
    result = python_udf.func(*python_udf.args, **python_udf.kwargs)
  File "/home/ubuntu/examples/distributed/rpc/ddp_rpc/main.py", line 48, in _run_trainer
    model = HybridModel(remote_emb_module, rank)
  File "/home/ubuntu/examples/distributed/rpc/ddp_rpc/main.py", line 30, in __init__
    self.fc = DDP(torch.nn.Linear(16, 8).cuda(device), device_ids=[device])
  File "/home/ubuntu/anaconda3/envs/pytorch_p38/lib/python3.8/site-packages/torch/nn/modules/module.py", line 688, in cuda
    return self._apply(lambda t: t.cuda(device))
  File "/home/ubuntu/anaconda3/envs/pytorch_p38/lib/python3.8/site-packages/torch/nn/modules/module.py", line 601, in _apply
    param_applied = fn(param)
  File "/home/ubuntu/anaconda3/envs/pytorch_p38/lib/python3.8/site-packages/torch/nn/modules/module.py", line 688, in <lambda>
    return self._apply(lambda t: t.cuda(device))
RuntimeError: CUDA error: invalid device ordinal
CUDA kernel errors might be asynchronously reported at some other API call,so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.

[W tensorpipe_agent.cpp:682] RPC agent for ps encountered error when reading incoming request from master: eof (this error originated at tensorpipe/transport/shm/connection_impl.cc:259)
[W tensorpipe_agent.cpp:682] RPC agent for trainer1 encountered error when reading incoming request from master: eof (this error originated at tensorpipe/transport/shm/connection_impl.cc:259)
[W tensorpipe_agent.cpp:682] RPC agent for trainer0 encountered error when reading incoming request from master: eof (this error originated at tensorpipe/transport/shm/connection_impl.cc:259)
[W tensorpipe_agent.cpp:682] RPC agent for ps encountered error when reading incoming request from trainer0: eof (this error originated at tensorpipe/transport/shm/connection_impl.cc:259)
[W tensorpipe_agent.cpp:682] RPC agent for ps encountered error when reading incoming request from trainer1: eof (this error originated at tensorpipe/transport/shm/connection_impl.cc:259)
Traceback (most recent call last):
  File "main.py", line 180, in <module>
    mp.spawn(run_worker, args=(world_size,), nprocs=world_size, join=True)
  File "/home/ubuntu/anaconda3/envs/pytorch_p38/lib/python3.8/site-packages/torch/multiprocessing/spawn.py", line 240, in spawn
    return start_processes(fn, args, nprocs, join, daemon, start_method='spawn')
  File "/home/ubuntu/anaconda3/envs/pytorch_p38/lib/python3.8/site-packages/torch/multiprocessing/spawn.py", line 198, in start_processes
    while not context.join():
  File "/home/ubuntu/anaconda3/envs/pytorch_p38/lib/python3.8/site-packages/torch/multiprocessing/spawn.py", line 160, in join
    raise ProcessRaisedException(msg, error_index, failed_process.pid)
torch.multiprocessing.spawn.ProcessRaisedException: 

-- Process 2 terminated with the following error:
Traceback (most recent call last):
  File "/home/ubuntu/anaconda3/envs/pytorch_p38/lib/python3.8/site-packages/torch/multiprocessing/spawn.py", line 69, in _wrap
    fn(i, *args)
  File "/home/ubuntu/examples/distributed/rpc/ddp_rpc/main.py", line 146, in run_worker
    fut.wait()
RuntimeError: RPCErr:1:RPC ran for more than set timeout (60000 ms) and will now be marked with an error
```

Based on looking at the code, it's because the [world size needs to be 4.](https://pytorch.org/docs/stable/rpc.html) Does this assume four separate machines that each map one process, or 4 distinct processes in one machine?  

Should I change the logic in [run_worker](https://github.com/pytorch/examples/blob/349c08e95acda442978e83400648afb3e3764036/distributed/rpc/ddp_rpc/main.py#L108) to assume one machine or is the intended behavior to work out of the box?

Reply:
world size of 4 means 4 separate processes which could either be on the same or different machines. They typically get gang scheduled so if one fails they all fail. So ideally `run_worker` shouldn't assume a local only scenario but if that's simplest for you to test I'm still happy merging the PR because it's not like we have tests for either case running now

Reply:
Just a quick update on this: I'm running these tests on 1.9 but encountering some issues that are taking a while to resolve, likely due to [these API updates]( https://discuss.pytorch.org/t/rpc-behavior-difference-between-pytorch-1-7-0-vs-1-9-0/124772/7). 

So far, I'm trying by tuning `run_worker` options like so, by tuning ulimits on my EC2 instance, and through increasing the timeout. 

`rpc_backend_options = TensorPipeRpcBackendOptions(_transports=["uv"])`

Reply:
It doesn't look like we can re-enable this test for 1.9 quite yet.

Reply:
That's ok maybe we merge your documentation change and we can have a seperate PR to enable the distributed test

Reply:
Sure, just for these correct `distributed/rpc/ddp_rpc/requirements.txt`? The other Makefile steps pass.

Reply:
yup that's right

Reply:
Ok, I reverted that specific change, let me know if it's ok to go otherwise. I can also file a new issue specific to this and assign it to myself. I'll use the previous issues as examples of how to write one.

Reply:
Created a new issue to follow up on this here https://github.com/pytorch/examples/issues/991